**🛑 CI red on required check — merge blocked pending test fix.**

Latest run (commit \`c032754d\`): \`test (ubuntu-latest, 22)\` **FAIL** (2m39s, 1 test failed of 4308).

**Failing test:** \`src/__tests__/metering.test.ts:600\` — \`MeteringService > getDailyTokenBreakdown > filters by from/to date range\`
\`\`\`
AssertionError: expected [] to have a length of 1 but got +0
 ❯ src/__tests__/metering.test.ts:600:22
\`\`\`

**Root cause:** Pre-existing time-bombed test, NOT introduced by this PR. Test was authored 2026-05-13 with hardcoded timestamps \`'2026-05-10'\` and \`'2026-05-12'\`. Production \`src/metering.ts:515-517\` (\`autoPrune\`) filters records older than \`Date.now() - maxAgeMs\`. Today is 2026-06-09 = exactly 28 days after the test's hardcoded date, so \`autoPrune\` correctly evicts the test records, leaving an empty array → \`toHaveLength(1)\` fails. Last green main CI was 2026-05-31 (PR #4536); no CI run on main between 2026-05-31 and 2026-06-09, so the regression went unnoticed.

**Branch protection:** \`test (ubuntu-latest, 22)\` is on the required-status-checks list for \`main\`. Per Boss (msg 1513850654420898003, 12:22 GMT+2): "Red gate = no merge to main, even docs-only. Default-to-Z covered opening the PR and backporting the link fix. It does not cover merging a red gate. Branch protection doesn't distinguish pre-existing vs new."

**Path to green** (per Boss, same message):
1. <@1490089546099069048> — fix \`metering.test.ts:600\` on \`develop\` (single-file, ~30 min, use \`vi.useFakeTimers()\` or \`Date.now() - N\` offsets). Tracked in #4637.
2. CI re-runs on PR #4636 (develop → main propagation via auto-refresh or rebase).
3. <@1490089830472880218> — formal review + LGTM (bot approval covers the 1-review requirement).
4. I push the merge button.

**My CI report was sloppy** — earlier in this thread I said "required checks all PASS" based on the first run; I should have re-verified after the second commit's CI run before reporting. Fixing that habit. Apologies.

**PR state is otherwise clean:**
- ✅ Link-integrity gap closed (3 files, 1 line changed + 419 lines of doc backports, all byte-identical to develop)
- ✅ Required checks passing: \`lint-pr-title\`, \`lint\`, \`Analyze\`
- ✅ \`feat-minor-bump-gate\` PASS
- ✅ \`CodeQL\`, \`Gitleaks\`, \`Trivy\`, \`GitGuardian\`, \`sdk-drift\`, \`Notify\`, \`Auto-label-test\`, \`dashboard-e2e\` (skipping on DRAFT-PR gate), \`platform-smoke\` (skipping on DRAFT-PR gate), \`test-matrix\` (skipping on DRAFT-PR gate)
- ❌ \`test (ubuntu-latest, 22)\` FAIL (pre-existing time-bomb, see #4637)
- ❌ \`test (ubuntu-latest, 20)\` FAIL (same root cause)
- ❌ \`helm-smoke\` FAIL (pre-existing k3d flakiness, NOT in required-check list, non-blocking)

**Unblocking:** Hep's \`fix/metering-timebomb-test\` PR to develop → CI re-runs PR #4636 → green → Argus LGTM → I merge.

Standing by. PR will sit open until #4637 lands on develop.

--- a/src/__tests__/metering.test.ts
+++ b/src/__tests__/metering.test.ts
@@ -563,6 +563,18 @@ describe('MeteringService', () => {
   });
   // ── getDailyTokenBreakdown (Issue #3282) ─────────────────────────
   describe('getDailyTokenBreakdown', () => {
+    beforeEach(() => {
+      // Disable auto-prune so backdated records survive the 30-day window.
+      metering.stop();
+      metering = new MeteringService(
+        eventBus,
+        (sessionId: string) => ownerMap.get(sessionId),
+        dataFile,
+        undefined,
+        { maxAgeMs: 0, maxRecords: 0 },
+      );
+    });
+
     it('returns empty array when no records exist', () => {
       expect(metering.getDailyTokenBreakdown()).toEqual([]);
     });

